spec/rspec/matchers/built_in/compound_spec.rb

```diff
@@ -375,6 +375,165 @@ def expect_block
             EOS
           end
         end
+
+        context "when the first matcher is diffable" do
+          subject { include("foo").and be_a(String) }
+
+          it 'is diffable' do
+            expect(subject).to be_diffable
+          end
+
+          context "when only first matcher fails" do
+            it 'fails with a message containing a diff for first matcher' do
+              allow(RSpec::Matchers.configuration).to receive_messages(:color? => false)
+
+              expected_failure = dedent(<<-EOS)
+                |Diff for (include "foo"):
+                |@@ -1,2 +1,3 @@
+                |-foo
+                |+baz
+                |+bar
+              EOS
+
+              expect {
+                expect(dedent(<<-EOS)).to subject
+                  |baz
+                  |bar
+                EOS
+              }.to fail_matching(expected_failure)
+            end
+          end
+
+          context "when only second matcher fails" do
+            subject { include("baz").and be_a(Fixnum) }
+
+            it 'fails with a message not containing a diff for first matcher' do
+              allow(RSpec::Matchers.configuration).to receive_messages(:color? => false)
+
+              expect {
+                expect(dedent(<<-EOS)).to subject
+                  |baz
+                  |bar
+                EOS
+              }.to fail_matching(a_string_excluding "Diff")
+            end
+          end
+
+          context "when both matcher fail" do
+            subject { include("foo").and eq(35) }
+
+            it 'fails with a message containing a diff with second matcher\'s failure' do
```

Comment:
* I'd prefer to use double quotes if you are using `'` as an apostrophe so we don't have to escape it.
* This doc string doesn't seem accurate...the diff shown below is for the 1st matcher's failure, not the 2nd matcher.